tools/nut-scanner/nut-scanner.c

```diff
@@ -305,6 +315,16 @@ int main(int argc, char *argv[])
 					timeout = DEFAULT_TIMEOUT*1000*1000;
 				}
 				break;
+                        case 'T' :
+                                thread_number = atol(optarg);
+                                if( thread_number == 0 ) {
+                                    fprintf(stderr, "Illegal thread number, using default %d\n", DEFAULT_THREAD);
+                                    thread_number = DEFAULT_THREAD;
+                                }
+#ifdef HAVE_PTHREAD
+                                sem_init(&semaphore, 0 , thread_number);
+#endif
+                                break;
```

Comment:
The help text says the default is DEFAULT_THREAD, but that's actually only applied if the -T option is given with an invalid number. It's unclear what the behavior should really be.